src/pages/Register/Register.tsx

```diff
@@ -38,14 +51,158 @@ const Register = ({
     }
   };
 
+  const navigateBack = () => {
+    navigate("/");
+  };
+
+  const navigate = useNavigate();
+
+  const handleFirstName = (event: FormEvent<HTMLInputElement>) => {
+    setFirstName(event.currentTarget.value);
+  };
+
+  const handleLastName = (event: FormEvent<HTMLInputElement>) => {
+    setLastName(event.currentTarget.value);
+  };
+
+  const isEmailValid = (email: string) => {
+    const emailRegex = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/;
+
+    return emailRegex.test(email);
+  };
+
+  const handleEmailInput = (event: FormEvent<HTMLInputElement>) => {
+    setEmail(event.currentTarget.value);
+  };
+
+  const beforeRegister = () => {
+    if (isEmailValid(email) == true && colorChange == false) {
+      setEmailColorChange(true);
+      return;
+    } else if (isEmailValid(email) == false) {
+      setEmailColorChange(false);
+    } else {
+      handleRegister();
+    }
+  };
+
+  const handlePasswordInput = (event: FormEvent<HTMLInputElement>) => {
+    setCheckConfirmPassword(event.currentTarget.value);
+    if (checkPassword === event.currentTarget.value) {
+      setPassword(event.currentTarget.value);
+      setColorChange(true);
+    } else {
+      setColorChange(false);
+    }
+  };
+
+  const handleClickNext = () => {
+    toggle();
+  };
+
+  const toggle = () => {
+    if (userinput) {
+      setUserInput(false);
+    } else {
+      setUserInput(true);
+    }
+  };
+
```

Comment:
can't you just change the toggle function to be called handleClickNext as all handleClickNext is doing is running the toggle function?

You can then just replace the one instance of where you are using the toggle function on and arrow image to handleClickNext.